### PR TITLE
feat(input): add files property

### PR DESCRIPTION
**Pull request by benelan**
**Related Issue:** #5890

## Summary
Adds the `files` property for `input type="file"`.

https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/files

### CODE REVIEW

**Review by eriklharper — APPROVED** [review]
Looks good to me, and if you want to attempt a test for this, perhaps this SO article could help out: https://stackoverflow.com/questions/68452480/test-file-upload-in-react-jest

File: src/components/input/input.tsx [hunk 1]
```diff
@@ -325,6 +325,13 @@ export class Input
   /** The component's value. */
   @Prop({ mutable: true }) value = "";
 
+  /**
+   * Contains the component's list of files when `type` is `"file"`.
```
Comment (benelan) [inline src/components/input/input.tsx:329]: @geospatialem can I get a quick doc review?

Comment (geospatialem) [reply src/components/input/input.tsx:329]: On it :)

Comment (geospatialem) [reply src/components/input/input.tsx:329]: WDYT about...

> When `type` is `"file"`, specifies the component's selected files.

**Comment by benelan** [comment]
I thought `files` was readonly and would only be changed when `value` changes but it looks like that was wishful thinking. This will need a bit more investigating

**Review by jcfranco — APPROVED** [review]
🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃
🗃📂🗃🗃🗃📂🗃🗃📂📂🗃🗃📂📂📂🗃🗃📂📂📂🗃📂📂📂📂🗃📂🗃
🗃📂📂🗃🗃📂🗃📂🗃🗃📂🗃🗃📂🗃🗃📂🗃🗃🗃🗃📂🗃🗃🗃🗃📂🗃
🗃📂🗃📂🗃📂🗃📂🗃🗃📂🗃🗃📂🗃🗃📂🗃🗃🗃🗃📂📂📂🗃🗃📂🗃
🗃📂🗃🗃📂📂🗃📂🗃🗃📂🗃🗃📂🗃🗃📂🗃🗃🗃🗃📂🗃🗃🗃🗃🗃🗃
🗃📂🗃🗃🗃📂🗃🗃📂📂🗃🗃📂📂📂🗃🗃📂📂📂🗃📂📂📂📂🗃📂🗃
🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃🗃

Before merging, can you add a test for this?

**Pull request merged**